app/Http/Controllers/EIAController.php

```diff
@@ -149,15 +149,19 @@ private function sendNotifications($project_ids)
                     if (stripos($project_location, $watcher->search) === false) continue;
                     if (isset($notifications_queue[$watcher->email][$project->id])) continue; // Only notify once for project per email
                     $hash = sha1($watcher->id . $watcher->search . $watcher->created_at);
-                    $project->setAttribute('unsubscribelinkloc', route('unsubscribe', [$watcher->email, $hash, $watcher->id]));
-                    $notifications_queue[$watcher->email][$project->id] = $project;
+                    $unsubscribelinkloc = route('unsubscribe', [$watcher->email, $hash, $watcher->id]);
+                    $notifications_queue[$watcher->email][$project->id] = [$project, $unsubscribelinkloc];
                     break; // Stop searching localities as watcher was already fulfilled
                 }
             }
         }
 
-        foreach ($notifications_queue as $email => $projects) {
-            $notifications = collect($projects);
+        foreach ($notifications_queue as $email => $projects_unsubscribelinklocs) {
+            $notifications = collect($projects_unsubscribelinklocs)
+                ->map(function ($project_unsubscribelinkloc) {
+                    $project_unsubscribelinkloc[0]->setAttribute('unsubscribelinkloc', $project_unsubscribelinkloc[1]);
+                    return $project_unsubscribelinkloc[0];
```

Comment:
@pato1 Ako `$project_unsubscribelinkloc[0]` a `$project_unsubscribelinkloc[1]` by som si asi najprv priradil do lokalnych premennych. Lebo takto s indexami to bude citatela v buducnosti miast.